addons/website_blog/static/src/js/blog_tag_add.js

```diff
@@ -0,0 +1,336 @@
+        this.$select = $(uiFragment.querySelector('[data-name="add_tag_opt"] we-select'));
+        this.tagIds.forEach(tag => {
+            this.$tagList.append(qweb.render('website_blog.tagListItem', {
+                tag: tag,
+                isEditing: this.isEditing,
+            }));
+        });
+        this.availableTags.forEach(tag => {
+            this.$select.prepend(qweb.render('website_blog.tagSelectItem', {
+                tag: tag,
+                isEditing: this.isEditing,
+            }));
+        });
+    },
+    _addTagPreview: function (tag) {
+        const $tagList = $('.tag_list');
+        const $newTag = $(qweb.render('website_blog.tag_link', {tag}));
+        $tagList.append($newTag);
+    },
+    _removeTagPreview: function (tag) {
+       const $removedTag = $(`.${tag.name}-${tag.id}`);
+       $removedTag.remove();
+    },
+    _saveChanges: function () {
+        const proms = [];
+        this.addList.forEach(async tag => {
+            if (tag.id === 0) {
+                proms.push(this._rpc({
+                    model: 'blog.tag',
+                    method: 'create',
+                    args: [{
+                        'name': tag.name,
+                        'post_ids': [this.blogPostId],
+                    }],
+                }));
+            } else {
+                proms.push(this._rpc({
+                    model: 'blog.tag',
+                    method: 'write',
+                    args: [tag.id, {
+                        'post_ids': [[4, this.blogPostId, 0]],
+                    }],
+                }));
+            }
+        });
+        this.unlinkList.forEach(tag => {
+            proms.push(this._rpc({
+            model: 'blog.tag',
+            method: 'write',
+            args: [tag.id, {
+                'post_ids': [[3, this.blogPostId, 0]],
+            }],
+            }));
+        });
+        this.addList = [];
+        this.unlinkList = [];
+        this.trigger_up('reset_tagList');
+        return Promise.all(proms);
+    },
+    _getTags: async function () {
+        this.tagIds = [];
+        this.availableTags = [];
+        let tags = await this._rpc({
+            model: 'blog.tag',
+            method: 'search_read',
+            args: [[], ['id', 'name', 'post_ids']],
+        });
+        tags.forEach(tag => {
+            if (tag.post_ids.includes(this.blogPostId)) {
+                this.tagIds.push(tag);
+            } else {
+                this.availableTags.push(tag);
+            }
+        });
+    },
+    _toggleEditUI: function () {
+        const $editMenu = $('[data-name="add_tag_opt"]');
+        $editMenu.toggleClass('d-none', !this.isEditing);
+        $('.o_we_edit_tag_btn').toggleClass('d-none', this.isEditing);
+        $('.o_we_save_tag_btn').toggleClass('d-none', !this.isEditing);
+
+    },
+});
+
+EditorMenuBar.include({
+    custom_events: Object.assign(EditorMenuBar.prototype.custom_events, {
+        set_tagList: '_onSetTags',
```

Comment:
Use more specific names. Any app could name an event 'set_tag_list' for its purpose